### PR TITLE
Redux

### LINKED ISSUE #19: Redux
After investigation, long-term wise it seems wiser to use Redux over Context - especially in regards to global state management and api calls

### PR DESCRIPTION
**Pull request by matthewdoles**

### CODE REVIEW

File: functions/quiz.js [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+  await Axios.get('https://opentdb.com/api_token.php?command=request')
+    .then(async (res) => {
+      if (res.data.response_code === 0) {
+        await Axios.get(
+          `https://opentdb.com/api.php?amount=10&encode=base64&category=${catId}&difficulty=${diff}&token=${res.data.token}`,
+        )
+          .then(async (response) => {
+            if (response.data.results.length === 10) {
+              triviaData = transformAPIData(response.data.results);
+            }
```
Comment (matthewdoles) [inline functions/quiz.js:16]: Only storing categories with redux. When creating the quiz I just make the callout and pass the data in navigation. Also just getting a new token everytime it makes a callout - probably a better way handle the refresh token but can work it out later.

File: functions/util.js [hunk 2]
```diff
@@ -8,3 +8,42 @@ export function shuffleArray(array) {
   }
   return array;
 }
+
+export function getTimeAgo(timestring) {
```
Comment (matthewdoles) [inline functions/util.js:12]: Instead of printing the date when a quiz was taken this gets how long ago it was

File: screens/Trivia/index.js [hunk 3]
```diff
@@ -8,41 +8,38 @@ import {
...
     console.log(categoryId, difficulty);
     setLoading(true);
-    await getTriviaData(categoryId, difficulty);
+    const triviaData = await getTriviaData(categoryId, difficulty);
     setLoading(false);
-    navigation.navigate(TRIVIAQ_SCREEN);
+    navigation.navigate(TRIVIAQ_SCREEN, { triviaData });
   };
 
   React.useLayoutEffect(() => {
```
Comment (quasney24) [inline screens/Trivia/index.js:45]: I know this isn't a change to this PR but i've always wondered what the purpose of this hook really was under the hood. Why would you use this vs a useEffect?

File: store/reducers/quizes.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const fetchQuizes = () => {
+  return async (dispatch, getState) => {
+    const { user } = getState();
+    dispatch({ type: FETCH_STATUS, isFetching: true });
+    await firebase
+      .firestore()
+      .collection('quizResults')
+      .where('uid', '==', user.user.uid)
```
Comment (quasney24) [inline store/reducers/quizes.js:42]: what happens if a user does not have quizResults collection yet? Would this fail out on the where clause?

File: screens/Trivia/index.js (hunk 3 above)
Comment (matthewdoles) [reply screens/Trivia/index.js:45]: useLayoutEffect runs synchronously and useEffect asynchronously. This just ensures the button override takes. useEffect might work, would have to try - but I think from trying in previous projects I was getting inconsistencies. Here is a resource that explains the differences pretty well: https://daveceddia.com/useeffect-vs-uselayouteffect/

File: store/reducers/quizes.js (hunk 4 above)
Comment (matthewdoles) [reply store/reducers/quizes.js:42]: Just logged in on as a test user and it's all good. Think it just returns an empty array - doesn't error out.

**Pull request merged**